R/projpred.R

```diff
@@ -194,6 +202,55 @@ get_refmodel.brmsfit <- function(object, newdata = NULL, resp = NULL,
         augdat_args_ilink = list(link = family$link)
       )
     }
+  } else if (latent) {
+    if (family$family == "cumulative") {
```

Comment:
this looks like a lot of code overhead just for the cumulative family. Why is this necessary? 

And if it is necessary, can we perhaps move it to its own function to keep the code here a bit cleaner? I would like special cases to be stored away tidily rather than being around like this in the main function.

Reply:
It's necessary to define argument `latent_ilink` which is a new argument within projpred, see [here](https://github.com/stan-dev/projpred/blob/543a7c3a93fcff12b2cf5fe3982323592ae4c729/R/extend_family.R#L26-L32) and [here](https://github.com/stan-dev/projpred/blob/543a7c3a93fcff12b2cf5fe3982323592ae4c729/R/extend_family.R#L131-L154). For the cumulative family, `latent_ilink` is basically `inv_link_cumulative()`, but the draws can be clustered or thinned (that's why I used `S_agg` in the inline comments).

Reply:
> And if it is necessary, can we perhaps move it to its own function to keep the code here a bit cleaner?

Yes, I can try to define `latent_ilink_tmp` outside of `get_refmodel.brmsfit()`. That will require handing over several objects (which is the reason why I didn't define it outside of `get_refmodel.brmsfit()` until now). I'll check how many and which objects are concerned.

Reply:
Thank you! Perhaps just passing the brms object (and some projpred stuff) would be sufficient?

Reply:
Done (commit https://github.com/paul-buerkner/brms/pull/1451/commits/0ebe5ef7c87171835603cc268876f3abdd9cf11f). Could you check whether my changes are ok?

> Perhaps just passing the brms object (and some projpred stuff) would be sufficient?

Unfortunately not. At least `resp` is required, too. And I also passed `family` and `bterms` to the new `get_latent_ilink_cumul()` function, for efficiency.

Reply:
I made some minor changes mainly due to my taste of code appearence. Can you check that I didn't break anything. Once I got your okay, I will merge this PR.

Reply:
Looks good, shouldn't break anything. I just noticed that you also removed the `rm(draws_mat)` statement. I guess this can lead to unnecessary memory usage (because `draws_mat` will continue to live in the environment of the returned function), but if you say this extra memory usage is ok, I'm fine with merging.